conda_smithy/feedstock_content/.gitignore

```diff
@@ -8,7 +8,7 @@
 !/conda-forge.yml
 
 # Don't ignore any subfolders if the parent folder is 'un-ignored'
-!/*/
+!/**/
 # Don't ignore any files/folders recursively in the following folders
 !/recipe/**
 !/.ci_support/**
```

Comment:
In https://github.com/conda-forge/webservices-dispatch-action/pull/41 you say:
> Should be unnecessary after [conda-forge/conda-smithy#1822](https://github.com/conda-forge/conda-smithy/pull/1822)

but I don't understand how `# Don't ignore any subfolders if the parent folder is 'un-ignored'` should affect `.github/workflows/*.yml` (the relevant files in https://github.com/conda-forge/webservices-dispatch-action/pull/41), since `.github` currently _is_ ignored.

Reply:
The actual error is 
```
.github/workflows
hint: Use -f if you really want to add them.
hint: Turn this message off by running
hint: "git config advice.addIgnoredFile false"
```
Adding `.github/workflows/*.yml` is fine, but the sub-folder `.github/workflows` is not. This PR fixes it.

Reply:
I don't see in the webservices repo [where](https://github.com/search?q=repo%3Aconda-forge%2Fwebservices-dispatch-action%20workflows&type=code) a bare `.github/workflows` would get added?

Also, I still don't see how this matches the comment. The base folder `.github` is ignored, so `parent folder is 'un-ignored'` does not apply. It's possible that the syntax is correct though, and that the comment is imprecise. The way I'd intuitively read the glob pattern is actually that _all_ non-folder files at least one level deep are un-ignored. It's kind of doubly inverted logic though, so I don't know what actually happens in practice.

Reply:
> It's kind of doubly inverted logic though, so I don't know what actually happens in practice.

It's a bit complicated, but I tried adding a bunch of different files and folders to see if the rules are correct and it seems like the rules are correct now IMO.

Reply:
So using the change from this PR:
```
cd numpy-feedstock
echo "asd" > .github/workflows/automerge.yml
git add .github/workflows/automerge.yml         # no warning
echo "asd" > .github/workflows/a.yml
git add .github/workflows/a.yml                 # WARNING; not added
mkdir .github/test
echo "asd" > .github/test/a.yml
git add .github/test/a.yml                      # WARNING; not added
rm -rf .github
git commit -m "remove .github"
mkdir -p .github/workflows
echo "asd" > .github/workflows/automerge.yml
git add .github/workflows/automerge.yml         # WARNING; not added
```

So empirically, `!/**/` seems to affect only(?) already checked-in files that would otherwise be gitignored. At least we should update/improve that comment.

Reply:
Try adding a file in `recipe/` and it will not work.

Reply:
> Try adding a file in `recipe/` and it will not work.

Not sure what you mean here. With `!/**/` committed in the local `.gitignore`, it worked fine to add a file in `recipe/`:
```
echo "asdf" > recipe/test.txt
git add recipe/test.txt  # no warning
git commit -m "done"
```

Reply:
I mean without that line, it will error.

Reply:
Nevermind, I get your point now; _without_ `!/**/`, adding a file to `recipe/` would also cause warnings.